gui/src/ruler.cc

```diff
@@ -0,0 +1,362 @@
+#include <cmath>
+#include <iostream>
+#include "ruler.hh"
+
+////////////////////////////////////////////////////////////////////////
+// Ruler
+
+Ruler::Ptr Ruler::create(Ruler::Orientation orientation, GtkWidget *drawingArea)
+{
+    Ruler::Ptr ruler{new Ruler(orientation)};
```

Comment:
Not sure why `clang-tidy` complains here, nor why it doesn't complain in many of the other places where we have similar constructs. The command-line version of `clang-tidy` seems fine with this, as is the CI build.

I would ignore this one for now. It'll sort itself out in time.